kiwi_keg/keg.py

```diff
@@ -37,7 +38,11 @@
     -d DEST_DIR, --dest-dir=DEST_DIR
         Destination directory for generated description, default cwd
 
-    -l, --list-recipes
+    --tar-overlays
+        Option to create a tarball root.tar.gz in destination directory
+        [default: false]
```

Comment:
Why would this be optional? Overlay is part of the image configuration, and as such the archives should be generated when keg generates the image description.

Reply:
After all the conversations we had, yes you are right the options needs to adapt. There is one situation which we need to take into account though. The "common" tarball can be either provided as "root.tar.gz" or as a directory named "root/...". I think the default for keg could be to create tarballs but there should be an option that allows to create the common root overlay as a directory structure. People who build with kiwi outside of obs never use a tarball for overlay files, they always use a directory structure which is much easier to manage than a binary blob.

Reply:
I didn't think of unpackaged root overlay. But, the question is do we even want to support it? I don't see a lot of value in that. Working with files in a tarball is a PITA for sure, but obviously with keg you can work with your files directly in your recipes environment. If you're meddling with overlay files directly in your kiwi build setup, you're probably not a keg user.

Reply:
As I said there is a value. Almost nobody who builds images outside of obs uses tarballs. A keg user that builds the image description for use outside of obs will use overlay files outside of a binary tarball, that is something I know from all customers I had contact with in the last years. Also in yesterdays conversation the idea was shared that we provide the keg written image descriptions in some public space. Also for this use case you want to provide files and not binary data.
The entire requirement for overlay files as tarballs is because of the open buildservice which can only deal with source files.

> If you're meddling with overlay files directly in your kiwi build setup, you're probably not a keg user.

I doubt that, for the reasons mentioned above.

Reply:
I think a good option is to change the creation of the tarballs to be true by default and the flag to disable it

thoughts ?

Reply:
sounds good yes. As Joe mentioned it can only be supported for the files that would go to root.tar.gz. Those files which are configured as ```<archive name="..."/>``` has to be in an archive.